numba/tests/test_unicode.py

```diff
@@ -362,6 +364,16 @@ def test_concat(self, flags=no_pyobj_flags):
                                  cfunc(a, b),
                                  "'%s' + '%s'?" % (a, b))
 
+    def test_repeat(self, flags=no_pyobj_flags):
+        pyfunc = repeat_usecase
+        cfunc = njit(pyfunc)
+        for a in UNICODE_EXAMPLES:
+            for b in (1, 2, 3):
+                self.assertEqual(pyfunc(a, b),
+                                 cfunc(a, b))
+                self.assertEqual(pyfunc(b, a),
+                                 cfunc(b, a))
+
```

Comment:
A test for the repeat exceptions would be good, e.g. non-integer type is not supported. You'll need to set `self.disable_leak_check()` in the test to make sure it ignores reference count leaks (as exceptions leak references!).

Reply:
makes sense, added in the latest commits.